src/core/qgsdataprovider.h

```diff
@@ -179,7 +179,7 @@ class CORE_EXPORT QgsDataProvider : public QObject
      *
      * \since QGIS 3.14
      */
-    virtual QgsDataProviderTemporalCapabilities *temporalCapabilities();
+    virtual QgsDataProviderTemporalCapabilities *temporalCapabilities() const;
```

Comment:
This shouldn't be const -- if you need a const getter, add a second one:

    virtual const QgsDataProviderTemporalCapabilities *temporalCapabilities() const;